src/modalities/models/gpt2/gpt2_model.py

```diff
@@ -226,20 +361,38 @@ def __init__(
         attention: AttentionConfig,
         activation: ActivationType,
         epsilon: float,
-        weight_init: WeightInitailizationConfig,
+        weight_init: WeightInitializationConfig,
     ):
         super().__init__()
         self.sample_key = sample_key
         self.prediction_key = prediction_key
         self.block_size = block_size
+        self.poe_type = poe_type
 
         assert vocab_size is not None
         assert block_size is not None
 
+        # TODO: dependency injection
+        if poe_type is PositionTypes.ABSOLUTE:
+            wpe = nn.Embedding(num_embeddings=block_size, embedding_dim=n_embd)
+        elif poe_type is PositionTypes.NOPE:
+            # Using a pre-trained layer, requires to define a separate FSDP unit for the frozen layer c.f.
+            # https://github.com/huggingface/accelerate/issues/807
+            # wpe = nn.Embedding.from_pretrained(torch.zeros(block_size, n_embd))
+            wpe = nn.Identity()
+        else:
+            raise TypeError(f"{poe_type} not supported")
+
+        if poe_type is PositionTypes.ABSOLUTE and RotaryTransform in [
+            config.type_hint.value for config in attention.qkv_transforms
+        ]:
+            logging.warning("You are using RotaryTransform together with absolute position embeddings."
+                            " It is expected to use \"RotaryTransform\" together with \"NOPE.\"")
```

Comment:
Should raise a ValueError. I don't see how having both could be useful.

Though, maybe we should add PositionTypes.ROTARY and in init add the RotaryTransform to qkv_transforms when that is set and it was not added yet.

Reply:
Done in 2ca59fb40d659cdb17b77275711eb4b1e45d34ab.